app/notification.go

```diff
@@ -30,10 +30,14 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 
 	pchan := a.Srv.Store.User().GetAllProfilesInChannel(channel.Id, true)
 	cmnchan := a.Srv.Store.Channel().GetAllChannelMembersNotifyPropsForChannel(channel.Id, true)
-	var fchan store.StoreChannel
-
+	var fchan chan store.StoreResult
 	if len(post.FileIds) != 0 {
-		fchan = a.Srv.Store.FileInfo().GetForPost(post.Id, true, true)
+		fchan = make(chan store.StoreResult, 1)
+		go func() {
+			fileInfos, err := a.Srv.Store.FileInfo().GetForPost(post.Id, true, true)
+			fchan <- store.StoreResult{Data: fileInfos, Err: err}
```

Comment:
Why pass as `Data` and `Err`?  Shouldn't those `fileInfos` and `err` be used directly?

Reply:
This is the approach that we are following for this PRs, using the gorouting + channel (we can't use directly the variables because are in gorouting so we need some way to sync it, and here we are syncing through the channels). After the whole campaing is completed, probably we move that approach to a WaitGroups based approach.